File: tournament_app/components/types/tournaments.ts
export interface Tournament {
  tournament_id: string;
  tournament_name: string;
  description: string;
  is_public: boolean;
  start_date: string;
  end_date: string;
  location: string;
  events: string[];
}

export interface PaginatedTournaments {
  items: Tournament[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

File: tournament_app_backend/app/models/documentModels/tournaments.py
from beanie import Document
from enum import Enum
from pydantic import Field
from beanie.odm.fields import PydanticObjectId
from datetime import date


class TournamentDBModel(Document):
    created_by: str = Field(title="Username")
    tournament_name: str = Field(title="Name of the tournament")
    description: str = Field(title="Description of the tournament")
    is_public: bool | None = Field(
        default=False, title="Specifies if the tournament is public or private"
    )
    start_date: date = Field(title="Start date of the tournament")
    end_date: date = Field(title="End date of the tournament")
    location: str = Field(title="Venue of the tournament")
    events: list[PydanticObjectId] = Field(title="Events of the tournament")

    class Settings:
        name = "tournament"
